src/lib.rs

```diff
@@ -1613,4 +1660,21 @@ r#"[
         let sum_by_col: Vec<u8> = grid.iter_cols().map(|col| col.sum()).collect();
         assert_eq!(sum_by_col, vec![1+4, 2+5, 3+6])
     }
+        #[test]
```

Comment:
Thx for adding tests. Could you please also add tests for out of bounds check? For example empty grid or idx greater size?